code/modules/spells/aoe_turf/conjure/pitbulls.dm

```diff
@@ -0,0 +1,56 @@
+/spell/aoe_turf/conjure/pitbull
+	name = "Summon Pitbulls"
+	desc = "This spell summons a group of misunderstood pitbulls."
+	user_type = USER_TYPE_WIZARD
+	specialization = SSOFFENSIVE
+
+	summon_type = list(/mob/living/simple_animal/hostile/pitbull)
+	summon_amt = 3
+
+	price = Sp_BASE_PRICE / 2
+	level_max = list(Sp_TOTAL = 2, Sp_SPEED = 2)
+	charge_max = 300
+	cooldown_reduc = 100
+	cooldown_min = 100
+	invocation = "GR'T W'TH K'DS"
+	invocation_type = SpI_SHOUT
+	spell_flags = NEEDSCLOTHES
+	override_icon = 'icons/mob/animal.dmi'
+	hud_state = "pitbull"  //TODO
+
+
+var/list/pitbulls_count_by_wizards = list()
+var/list/pitbulls = list()
+#define MAX_PITBULLS 15
+
+/spell/aoe_turf/conjure/pitbull/cast_check(skipcharge = 0,mob/user = usr)
+	if (pitbulls_count_by_wizards[user] > MAX_PITBULLS) // We summoned too many pitbulls :(
+		to_chat(user, "<span class = 'warning'>We've brought forth too many innocent pitbulls into this plane.</span>")
+		return FALSE
+	return ..()
+
+
+/spell/aoe_turf/conjure/pitbull/summon_object(var/type, var/location)
+	..()
+	pitbulls_count_by_wizards[holder]++
```

Comment:
the wizard is never removed from this global list, so it will never be properly deleted.
either clear it manually or use weak references through `\ref`

Reply:
@DamianX I might need some help implementing the weak reference, I have no idea how to even start.

Reply:
Actually, weak references are stupid.
Just do
```
if(pitbull_count_by_wizard[my_wiz] == 0)
    pitbulls_count_by_wizard -= my_wiz
```
or something like that after you decreased the count in `pitbull/death()`

Reply:
@DamianX 
I already have `pitbulls_count_by_wizards[my_wiz]--` in `pitbull/death()`, is that sufficient?

Reply:
oh wait I misunderstood, yeah I gotta delete the list when it's empty ignore that last comment

Reply:
You not-so-much have to delete the list, it's more that you have to remove the reference to the wizard from the list, at least.